integrate.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2013 Michael Mims <mims.michael@gmail.com>
+ * Copyright 2014 Martin Pöhlmann <martin.deimos@gmx.de>
+ * Copyright 2014 Jiří Janoušek <janousek.jiri@gmail.com>
+ * Copyright 2015 Aaron Cripps <acripps@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+"use strict";
+
+(function(Nuvola)
+{
+	// Create media player component
+	var player = Nuvola.$object(Nuvola.MediaPlayer);
+
+	// Handy aliases
+	var PlaybackState = Nuvola.PlaybackState;
+	var PlayerAction = Nuvola.PlayerAction;
+
+	// Create new WebApp prototype
+	var WebApp = Nuvola.$WebApp();
+
+	// Initialization routines
+	WebApp._onInitWebWorker = function(emitter)
+	{
+		Nuvola.WebApp._onInitWebWorker.call(this, emitter);
+
+		var state = document.readyState;
+		if (state === "interactive" || state === "complete")
+			this._onPageReady();
+		else
+			document.addEventListener("DOMContentLoaded", this._onPageReady.bind(this));
+	}
+
+	WebApp._onPageReady = function()
+	{
+		// Connect handler for signal ActionActivated
+		Nuvola.actions.connect("ActionActivated", this);
+
+		// Start update routine
+		this.update();
+	}
+
+	// Extract data from the web page
+	WebApp.update = function()
+	{
+		var can_prev = false;
```

Comment:
It would be better to use JavaScript naming convention for variables: `canPrev`, `canPlay`, etc. I admit the coding style in Nuvola Player 2 is not good, but now we have a chance to improve it :-)